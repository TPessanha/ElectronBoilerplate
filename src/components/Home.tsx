import React from "react";
import * as style from "./HomeStyle.scss";

export interface IHomeProps {
	compiler: string;
	framework: string;
}

// 'IHomeProps' describes the shape of props.
// State is never set so we use the '{}' type.
export class Home extends React.Component<IHomeProps, {}> {
	public render() {
		return (
			<div>
				<h1 className={style.test}>
					Hello from {this.props.compiler} and {this.props.framework}!
				</h1>
			</div>
		);
	}
}
