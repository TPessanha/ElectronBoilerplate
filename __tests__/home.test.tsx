import React from "react";
import renderer from "react-test-renderer";
import { Home } from "../src/components/Home";

it("renders home correctly", () => {
	const tree = renderer
		.create(<Home compiler="TypeScript" framework="React" />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
