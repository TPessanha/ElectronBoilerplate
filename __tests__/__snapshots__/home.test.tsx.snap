// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home correctly 1`] = `
<div>
  <h1>
    Hello from 
    TypeScript
     and 
    React
    !
  </h1>
</div>
`;
